var/spack/repos/spack_repo/builtin/packages/strumpack/package.py

```diff
@@ -104,6 +104,8 @@ class Strumpack(CMakePackage, CudaPackage, ROCmPackage):
         depends_on(
             "slate amdgpu_target={0}".format(val), when="+slate amdgpu_target={0}".format(val)
         )
+    for val in CudaPackage.cuda_arch_values:
+        depends_on("slate cuda_arch={0}".format(val), when="+slate cuda_arch={0}".format(val))
```

Comment:
You can "tighten" this up a bit using f-strings:

```suggestion
        depends_on(f"slate cuda_arch={val}", when=f"+slate cuda_arch={val}")
```

However, shouldn't the condition also required `+cuda`?

Reply:
Addressed in https://github.com/spack/spack/pull/50460/commits/7795358a7658c9ff43adc3196b97b86b1ce81c1c, went in favour of `when("+cuda+slate")` in other comment.